File: apps/backend/src/adapters/inbound/http/dto/service_record.rs
use chrono::NaiveDate;
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateServiceRecordRequest {
    pub service_date: NaiveDate,
    pub odometer: u32,
    pub description: String,
    pub workshop: Option<String>,
    pub cost: Option<Decimal>,
    pub notes: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct ServiceRecordResponse {
    pub id: String,
    pub vehicle_id: String,
    pub service_date: NaiveDate,
    pub odometer: u32,
    pub description: String,
    pub workshop: Option<String>,
    /// Serialised as a string to avoid floating-point precision loss.
    pub cost: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct ServiceRecordListResponse {
    pub records: Vec<ServiceRecordResponse>,
}

File: apps/mobile/lib/features/service_record/domain/entities/service_record.dart
import 'package:decimal/decimal.dart';

/// Domain entity — pure Dart. The data layer maps a DTO into this.
class ServiceRecord {
  const ServiceRecord({
    required this.id,
    required this.vehicleId,
    required this.serviceDate,
    required this.odometer,
    required this.description,
    this.workshop,
    this.cost,
    this.notes,
  });

  final String id;
  final String vehicleId;
  final DateTime serviceDate;
  final int odometer;
  final String description;
  final String? workshop;
  final Decimal? cost;
  final String? notes;
}
